### PR TITLE
create rup architecture doc

### PR DESCRIPTION
**Pull request by jens1o**

### CODE REVIEW

File: docs/RUP_Architecture.md [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+
+For each significant class in the package, include its name, brief description, and, optionally a description of some of its major responsibilities, operations and attributes.]
+
+### Process View
+
+[This section describes the system's decomposition into lightweight processes (single threads of control) and heavyweight processes (groupings of lightweight processes). Organize the section by groups of processes that communicate or interact. Describe the main modes of communication between processes, such as message passing, interrupts, and rendezvous.]
+
+### Deployment View
+
+[This section describes one or more physical network (hardware) configurations on which the software is deployed and run. It is a view of the Deployment Model. At a minimum for each configuration it should indicate the physical nodes (computers, CPUs) that execute the software, and their interconnections (bus, LAN, point-to-point, and so on.) Also include a mapping of the processes of the Process View onto the physical nodes.]
```
Comment (Jonas-Karl) [inline docs/RUP_Architecture.md:93]: Will be added in the future

File: docs/RUP_Architecture.md [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+The backend (Django based) and the frontend are both developed separted from each other and only communicate over a REST API. That said, it is hard to follow one of the known patterns (MVC, MVP, MVVM) for the whole project, as the frontend ecosystem (an Android app) can only be so much used to follow one of some known pattterns.
+
+Please also see this [introduction to the technologies](https://fibo952390745.wordpress.com/2022/10/18/week-2-roles-and-technologies/).
+
+![Overview of the Architecture](architectural_UML_Diagram.drawio.svg "Architecture")
+
+### Architectural Goals and Constraints
+
+[This section describes the software requirements and objectives that have some significant impact on the architecture, for example, safety, security, privacy, use of an off-the-shelf product, portability, distribution, and reuse. It also captures the special constraints that may apply: design and implementation strategy, development tools, team structure, schedule, legacy code, and so on.]
+
```
Comment (Jonas-Karl) [inline docs/RUP_Architecture.md:60]: Safety and maybe portablility: we use Tokens to handle user sessions which also affects the Architecture

File: docs/RUP_Architecture.md [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+### Use-Case View
+
+[This section lists use cases or scenarios from the use-case model if they represent some significant, central functionality of the final system, or if they have a large architectural coverage - they exercise many architectural elements, or if they stress or illustrate a specific, delicate point of the architecture.]
+
+#### Use-Case Realizations
+
+[This section illustrates how the software actually works by giving a few selected use-case (or scenario) realizations, and explains how the various design model elements contribute to their functionality.]
+
```
Comment (Jonas-Karl) [inline docs/RUP_Architecture.md:68]: Maybe we could mention the .feature files here

Comment (Jonas-Karl) [inline docs/RUP_Architecture.md:68]: Should we also include the Mockups here?

**Review by ChrisEwert — APPROVED** [review]
Super

File: docs/RUP_Architecture.md (hunk 3 above)
Comment (Cebox82) [reply docs/RUP_Architecture.md:68]: for me it is not important, because we have no working cucumber test yet either

Comment (Cebox82) [reply docs/RUP_Architecture.md:68]: if it is not necesary then rather not. After all, we have them in the repo
